base/sort.jl

```diff
@@ -164,6 +164,11 @@ same thing as `partialsort!` but leaving `v` unmodified.
 partialsort(v::AbstractVector, k::Union{Integer,OrdinalRange}; kws...) =
     partialsort!(copymutable(v), k; kws...)
 
+function midpoint(lo::T, hi::T) where T<:Integer
+    l, h = div(lo, 2), div(hi, 2)
+    return l + h + ifelse(isodd(lo) & isodd(hi) & !(signbit(lo) ⊻ signbit(hi)), sign(lo), zero(lo))
```

Comment:
😱

Reply:
Should I add a comment? :smiling_imp: It's actually pretty simple: `div` rounds down; if both are odd and have the same sign, you need to bump the result by one in the proper direction.